File: 08_OOP_Python_Eissorten/fruchteis.py
from eissorte import Eissorte

# Klasse Fruchteis erbt von / erweitert Eissorte
# Python kennt kein "extends"
class Fruchteis(Eissorte):

    def __init__(self, sorte, farbe, preis, ist_verfuegbar, fruchtanteil, ist_vegan):
        super().__init__(sorte, farbe, preis, ist_verfuegbar)
        self.fruchtanteil = fruchtanteil
        self.ist_vegan = ist_vegan

    def gib_informationen_aus(self):
        super().gib_informationen_aus()
        print(f"Fruchtanteil: {self.fruchtanteil}")
        print(f"Veagn: {self.ist_vegan}")
        print()

File: 07_OOP/src/Fruchteis.java
public class Fruchteis extends Eissorte implements Schmelzbar {
    private int fruchtAnteil;
    private boolean istVegan;

    public Fruchteis(String sorte, int preis, boolean istVerfuegbar, String farbe, int fruchtAnteil, boolean istVegan) {
        super(sorte, preis, istVerfuegbar, farbe);
        this.fruchtAnteil = fruchtAnteil;
        this.istVegan = istVegan;
    }

    @Override
    protected void gibInformationenAus() {
        //super.gibInformationenAus();
        System.out.println("Ich bin die Sorte " + super.sorte + ", koste " +
                super.formatierePreis() + " bin " + super.farbe);
        System.out.println("Zusätzlich habe ich einen Fruchtanteil von " + this.fruchtAnteil +
                " und bin vegan: " + this.istVegan);
    }

    @Override
    public void schmelzen() {
        System.out.println(getSorte() + " der Klasse Fruchteis schmilzt");
    }
}
